content/code-signing/ios-code-signing.md

```diff
@@ -34,45 +34,63 @@ For successful signing, the certificate and the provisioning profile must match
 
 {{<notebox >}} 
-The actions that Codemagic can perform on your behalf depend on your [Apple Developer Program user role](https://help.apple.com/app-store-connect/#/deve5f9a89d7).
+Note that users how had previously set up the session-based intgeration have been automatically migrated to use the API-key based setup.
 {{</notebox>}}
 
+1. In the list of available integrations, click the **Connect** button for **Developer Portal**.
+2. In the **App Store Connect API key name**, provide a name for the key you are going to set up the integration with. This is for identifying the key in Codemagic.
+3. Enter the **Issuer ID** related to your Apple Developer account. You can find it above the table of active keys on the Keys tab of the [Users and Access](https://appstoreconnect.apple.com/access/api) page.
+4. Enter the **Key ID** of the key to be used for code signing.
+5. In the **API key** field, upload a RSA 2048 bit private key to be included in the [signing certificate](https://help.apple.com/xcode/mac/current/#/dev1c7c2c67d) that Codemagic creates. You can use an existing key or create a new 2048 bit RSA key by running the following command in your terminal:
+
+        ssh-keygen -t rsa -b 2048 -f ~/Desktop/codemagic_private_key -q -N ""
+
+6. Click **Save** to finish the setup.
+
+If you work with multiple Apple Developer teams, you can add additional keys by clicking **Add another key** right after adding the first key and repeating the steps described above. You can delete existing keys or add new ones when you click **Manage keys** next to the Developer Portal integration in user or team settings.
+
+### Step 3. Enabling automatic code signing for workflow
+
+Once the Apple Developer Portal has been enabled for the account or team the app belongs to, you can easily enable automatic code signing per workflow.
+
 1. Go to **App settings > Publish > iOS code signing**.
 2. Select **Automatic** as the code signing method. If you haven't enabled the Apple Developer Portal integration yet, you will be asked to enable it before you can continue configuration.
-3. If you belong to several Apple Developer teams, select the right team in the **Developer portal team** field.
-4. Select the **provisioning profile type** used for provisioning the build. Codemagic will automatically select or generate a matching certificate for code signing.
+3. If you have several keys available, select the right key in the **App Store Connect API key** field.
+4. Select the **provisioning profile type** used for provisioning the build. Codemagic will automatically select or generate a matching certificate for code signing. The provisioning profiles (except for Distribution) will include all the devices you have registered on your Apple Developer account.
```

Comment:
Maybe it should be noted that when provisioning profiles are created, then they include all the devices available at the time. Later when profiles are re-used, new devices will not be included in them.

Reply:
Updated